system/jlib/jsocket.cpp

```diff
@@ -7194,3 +7238,18 @@ extern jlib_decl void shutdownAndCloseNoThrow(ISocket * optSocket)
         e->Release();
     }
 }
+
+////
+void ScopedNonBlockingMode::init(ISocket *_socket)
+{
```

Comment:
I think we should add here -
+    if (connectionless())
+        return;

Reply:
I don't think it's incorrect at the moment.
It could make sense for UDP readtms/writetms to call and return blocked before it could read anything,
and not that we use it anywhere, if it was UDP+SSL, I suspect it would be necessary too.

Saying that, in our current use patterns, I don't think it's useful (we didn't need it before, so we don't need it now?), so I agree, to minimize the impact, we'll switch off the non-blocking mode for UDP for now.

Reply:
I just realized - if we turn off non-block for UDP then we must put the wait_read() back into the beginning of readtms() for UDP - before the recvfrom.

Reply:
Perhaps we could just change recv() and send() to use MSG_DONTWAIT flag ?
So we dont have issue with one thread in reading and one thread writing and getting non-blocking flag mixed up ?
(but still need to decide about UDP - if it gets wait_read() added back in or we also use MSG_DONTWAIT))

Reply:
@jakesmith I think what remains for this PR is:
1). UDP - if we decide to leave as blocking - put wait_read() into jsocket readtms() / wait_write() into writetms().
2). set all TCP sockets as non-blocking after connect()/accept() or leave in readtms()/writetms(), but don't restore to blocking, just leave socket in non-blocking mode always.